src/app/Shared/Selector.component.tsx

```diff
@@ -75,68 +73,32 @@ const selectStyle: StylesConfig = {
     }),
 };
 
-/**
- * @desc Transforms a string array to an object.
- * @param {string[]} selectorArray - array with selector values
- * @returns {Record<string, string>[]} - object with value und label from the string array
- */
-function generateSelectorObject(
-    selectorArray: string[],
-    translateFunction: TFunction,
-    isFeature: boolean
-): Record<string, string>[] {
-    const selectorObject: Record<string, string>[] = [];
-    selectorArray.forEach((element) => {
-        let label = element;
-        if (isFeature) {
-            label = translateFunction(`Filters.${element}`);
-        }
-        selectorObject.push({ value: element, label });
-    });
-    return selectorObject;
-}
-
 export interface SelectorProps {
     label: string;
-    selectValues: string[];
+    dropDownValuesObj: { value: string; label: string }[];
+    selectedValuesObj: { value: string; label: string }[];
```

Comment:
question: is the selectedValues a slice of the data of the dropDownValues? If so it would be sufficient to give the selectedValues as an value[] where the values refer to the values in the dropDownValuesObj.

Reply:
Yes they are. But the selector package expects an ObjectList also for selectedValues.

Reply:
Ok, if you feel like, you can move the selecetedValuesObj generating code here, so that the selectedValuesProps is only a value[]. But perhaps sometimes you want the selected labels to be different than the initial labels, so you can also leave it as it is.

Reply:
I think it's okay the way it is for now.